traveler-ios-ui/TravelerKitUI/ViewControllers/CancellationViewController.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  CancelOrderViewController.swift
+//  TravelerKitUI
+//
+//  Created by Omar Padierna on 2019-06-04.
+//  Copyright © 2019 GuestLogix. All rights reserved.
+//
+
+import UIKit
+import TravelerKit
+
+protocol CancellationViewControllerDelegate: class {
+    func cancellationViewController(_ controller: CancellationViewController, didCancel order:Order)
+    func cancellationViewController(_ controller: CancellationViewController, cancelationDidFail error: Error)
+}
+
+class CancellationViewController: UITableViewController {
+    @IBOutlet weak var totalRefundLabel: UILabel!
+
+    var quote: CancellationQuote?
+    weak var delegate:CancellationViewControllerDelegate?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        totalRefundLabel.text = quote?.totalRefund.localizedDescription
+    }
+
+    // MARK: UITableViewDataSource
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return quote?.products.count ?? 0
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "productCancelCell", for: indexPath) as! InfoCell
+
+        cell.titleLabel.text = quote?.products[indexPath.row].title
+        let percentageFee = (quote?.cancellationCharge.value ?? 0.0/quote!.products[indexPath.row].totalRefund.value ) * 100.0
```

Comment:
you can force unwrap here too :)

Reply:
The calculation of the percentage (is business rule) and should live inside an extension on the model maybe?